src/test/regress/expected/multi_router_planner.out

```diff
@@ -1385,6 +1385,24 @@ DEBUG:  Plan is router executable
 --------------+-------------------
 (0 rows)
 
+-- is null clause is not shard prunable so, not router plannable
```

Comment:
Please add an introduction sentence explaining why these two tests are needed, maybe "the following queries are equivalent, but only the one with an equality constraint is shard-prunable, meaning only it can produce a router plan. The first will hit all shards and the second will hit no shards. Both should return no rows"

That's a lot of information, but looking at these tests I couldn't quite figure out why they'd been added. Maybe add some subset of the above to help future readers understand what is being tested (your existing comments just explain what is expected, but not why).